.github/workflows/limit-pr-files-changed.yml

```diff
@@ -0,0 +1,87 @@
+name: Enforce Max Lines Changed Per File
+
+on:
+  pull_request:
+    paths:
+      - '**/*.java'   # Monitor changes to Java files
+      - '**/*.avsc'   # Monitor changes to Avro schema files
+      - '**/*.proto'   # Monitor changes to proto schema files
+
+jobs:
+  enforce-lines-changed:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get PR Metadata
+        id: pr_details
+        uses: actions-ecosystem/action-get-PR@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check for Override Keyword
+        id: check_override
+        run: |
+          # Define the keyword for override
+          OVERRIDE_KEYWORD="DIFFSIZEOVERRIDE"
+          
+          # Check PR title and body for the keyword
+          if echo "${{ steps.pr_details.outputs.title }}${{ steps.pr_details.outputs.body }}" | grep -iq "$OVERRIDE_KEYWORD"; then
+            echo "Override keyword found. Skipping validation."
+            echo "override=true" >> $GITHUB_OUTPUT
+          else
+            echo "override=false" >> $GITHUB_OUTPUT
+          fi
+      - name: Calculate Lines Changed Per File
+        if: ${{ steps.check_override.outputs.override != 'true' }}
+        id: lines_changed_per_file
+        run: |
+          # Define the maximum allowed lines changed per file
+          MAX_LINES=500
+
+          # Get the diff of the PR and process each file
+          EXCEEDED=false
+          JAVA_FILE=false
+          SCHEMA_FILE=false
+          while IFS=$'\t' read -r added removed file; do
+          # Skip test files
+          if [[ "$file" != *src/main* ]]; then
+            echo "Skipping file: $file"
+            continue
+          fi
+          
+          if [[ "$file" == *.java ]]; then
+            JAVA_FILE=true
+          else
+            SCHEMA_FILE=true
+          fi
+
+          # Calculate total lines changed for the file
+          TOTAL=$((added + removed))
+          echo "File: $file, Lines Changed: $TOTAL"
+
+          # Check if the total exceeds the max allowed
+          if [ "$TOTAL" -gt "$MAX_LINES" ]; then
+            echo "File $file exceeds the maximum allowed lines changed ($TOTAL > $MAX_LINES)"
+            EXCEEDED=true
+          fi
+          done < <(git diff --numstat upstream/main | grep -E '\.(java|avsc|proto)$')
```

Comment:
Just double check, you changed the target branch from origin to upstream, is it intended?

Reply:
changed back to origin